pkg/datastore/filedb/filedb.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2022 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filedb
+
+import (
+	"context"
+	"encoding/json"
+
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipecd/pkg/datastore"
+	"github.com/pipe-cd/pipecd/pkg/filestore"
+)
+
+type FileDB struct {
+	backend filestore.Store
+	logger  *zap.Logger
+}
+
+type Option func(*FileDB)
+
+func WithLogger(logger *zap.Logger) Option {
+	return func(m *FileDB) {
+		m.logger = logger
+	}
+}
+
+func NewFileDB(fs filestore.Store, opts ...Option) (*FileDB, error) {
+	fd := &FileDB{
+		logger: zap.NewNop(),
+	}
+	for _, opt := range opts {
+		opt(fd)
+	}
+
+	fd.backend = fs
+
+	return fd, nil
+}
+
+func (f *FileDB) Find(ctx context.Context, kind string, opts datastore.ListOptions) (datastore.Iterator, error) {
+	return nil, datastore.ErrUnimplemented
+}
+
+func (f *FileDB) Get(ctx context.Context, kind, id string, v interface{}) error {
+	// TODO: Find a better way to avoid knowledge leak about Application kind.
+	var path string
+	switch kind {
+	case datastore.ApplicationModelKind:
+		return datastore.ErrUnimplemented
```

Comment:
Instead of sticking with the model name like this, we can define some properties each model could have.
For examples:
- Divide to multiple files in hot storage or not
- Use cold storage or not

Then the implementation of FileDB simply relies on those properties instead of each model kind.

Reply:
Ah, I get your point. That's nice, we need kind of interface which all models should implement like
```go
type Model interface {
  DivideToMulti() []string
  StoreAsCold() bool
}
```
I mean, instead of using object properties, an interface may be a better choice due to the current datastore interface using `v interface{}` as a parameter.

Reply:
Yes. Looks nice.

Reply:
Thanks for the idea, I will make a separated PR for that first, just don't want to make too many changes in one PR  🙏